pkg/apis/performanceprofile/v2/performanceprofile_validation_test.go

```diff
@@ -369,6 +369,62 @@ var _ = Describe("PerformanceProfile", func() {
 			})
 		})
 	})
+
+	Describe("validation of validateFields function", func() {
+		It("should check all fields", func() {
+			// config all specs to rise an error in every func inside validateFields()
+			reservedCPUs := CPUSet("")
+			isolatedCPUs := CPUSet("0-6")
+			offlinedCPUs := CPUSet("7")
+			profile.Spec.CPU.Reserved = &reservedCPUs
+			profile.Spec.CPU.Isolated = &isolatedCPUs
+			profile.Spec.CPU.Offlined = &offlinedCPUs
+
+			profile.Spec.MachineConfigLabel["foo"] = "bar"
+
+			incorrectDefaultSize := HugePageSize("!#@")
+			profile.Spec.HugePages.DefaultHugePagesSize = &incorrectDefaultSize
+
+			profile.Spec.WorkloadHints = &WorkloadHints{
+				RealTime: pointer.BoolPtr(false),
+			}
+			profile.Spec.RealTimeKernel = &RealTimeKernel{
+				Enabled: pointer.BoolPtr(true),
+			}
+
+			invalidVendor := "123"
+			invalidDevice := "0x12345"
+			profile.Spec.Net.Devices[0].InterfaceName = pointer.StringPtr("")
+			profile.Spec.Net.Devices[0].VendorID = pointer.StringPtr(invalidVendor)
+			profile.Spec.Net.Devices[0].DeviceID = pointer.StringPtr(invalidDevice)
+
+			errors := profile.validateFields()
+
+			errorMsgs := map[string]bool{}
+			errorMsgs["reserved CPUs can not be empty"] = false
+			errorMsgs["you should provide only 1 MachineConfigLabel"] = false
+			errorMsgs[`hugepages default size should be equal to "1G" or "2M"`] = false
+			errorMsgs["device name cannot be empty"] = false
+			errorMsgs[fmt.Sprintf("device vendor ID %s has an invalid format. Vendor ID should be represented as 0x<4 hexadecimal digits> (16 bit representation)", invalidVendor)] = false
+			errorMsgs[fmt.Sprintf("device model ID %s has an invalid format. Model ID should be represented as 0x<4 hexadecimal digits> (16 bit representation)", invalidDevice)] = false
+			errorMsgs["realtime kernel is enabled, but realtime workload hint is explicitly disable"] = false
+
+			for _, err := range errors {
+				if !errorMsgs[err.Detail] {
+					errorMsgs[err.Detail] = true
+				}
+			}
+
+			testIsCorrect := true
+			for _, element := range errorMsgs {
+				if !element {
+					testIsCorrect = false
+				}
+			}
+
+			Expect(testIsCorrect).To(BeTrue())
```

Comment:
let's have an annotated assetion to make it obvious which error is missing. We can do in many ways. One of them is
```
for errorMsg, ok := range errorMsgs {
    Expect(ok).To(BeTrue(), "missing expected error: %q", errorMsg)
}
```